code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -901,26 +918,19 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 /datum/reagents/proc/is_full()
 	return total_volume >= maximum_volume
 
-/datum/reagents/proc/get_heatcapacity()
-	var/heat_capacity = 0
-
-	if(reagent_list.len)
-		for(var/datum/reagent/R in reagent_list)
-			heat_capacity += R.volume*R.specheatcap
-
-	return heat_capacity
-
 /datum/reagents/proc/get_overall_mass()
 	//M = DV
-
 	var/overall_mass = 0
-
-	if(reagent_list.len)
-		for(var/datum/reagent/R in reagent_list)
-			overall_mass += R.density*R.volume
-
+	for(var/datum/reagent/R in reagent_list)
+		overall_mass += R.density*R.volume
 	return overall_mass
 
+/datum/reagents/proc/get_thermal_mass()
+	var/total_thermal_mass = 0
+	for(var/datum/reagent/R in reagent_list)
+		total_thermal_mass += R.volume * R.density * R.specheatcap
```

Comment:
i thought specheatcap was joules per unit volume, not mass?
do we even use reagent density for anything else than just immediately multiplying it with the specheatcap

Reply:
Specheatcap is per unit mass.

Reply:
On the one hand, by volume would make more sense in terms of code design, like how for gases it's by mole
On the other hand, by mass is used much more often in reality